src/flamenco/runtime/program/fd_stake_program.c

```diff
@@ -1308,6 +1324,8 @@ delegate( fd_exec_instr_ctx_t const *   ctx,
           fd_pubkey_t const *           signers[static FD_TXN_SIG_MAX] ) {
   int rc;
 
+  fd_valloc_t scratch_valloc = fd_scratch_virtual();
```

Comment:
Should we wrap the whole instruction processor in a `FD_SCRATCH_SCOPE`? To make the semantics clear?